core/src/test/java/com/graphhopper/routing/util/spatialrules/AreaIndexTest.java

```diff
@@ -0,0 +1,182 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.graphhopper.routing.util.spatialrules;
+
+import com.graphhopper.routing.util.AreaIndex;
+import com.graphhopper.routing.util.CustomArea;
+import com.graphhopper.util.StopWatch;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.geom.LinearRing;
+import org.locationtech.jts.geom.Polygon;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.graphhopper.util.GHUtility.readCountries;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+class AreaIndexTest {
+
+    @Test
+    void basic() {
+        GeometryFactory geometryFactory = new GeometryFactory();
+        Polygon border1 = geometryFactory.createPolygon(new Coordinate[]{
+                new Coordinate(1, 1), new Coordinate(2, 1), new Coordinate(2, 2), new Coordinate(1, 2),
+                new Coordinate(1, 1)});
+        Polygon border2 = geometryFactory.createPolygon(new Coordinate[]{
+                new Coordinate(5, 5), new Coordinate(6, 5), new Coordinate(6, 6), new Coordinate(5, 6),
+                new Coordinate(5, 5)});
+        Polygon border3 = geometryFactory.createPolygon(new Coordinate[]{
+                new Coordinate(9, 9), new Coordinate(10, 9), new Coordinate(10, 10), new Coordinate(9, 10),
+                new Coordinate(9, 9)});
+        AreaIndex<CustomArea> index = new AreaIndex<>(Arrays.asList(
+                createCustomArea("1", border1),
+                createCustomArea("2", border2),
+                createCustomArea("3", border3),
+                createCustomArea("4", border2, border3)));
+        testQuery(index, 0, 0);
+        testQuery(index, 1.5, 1.5, "1");
+        // todo: is it somehow necessary that coordinates right on the border are included?
```

Comment:
I'm not sure we'll ever face problems with this one in reality. JTS uses full double-precision for its checks(See [FAQ](https://locationtech.github.io/jts/jts-faq.html#C3)) and the chance that the center of a way is *exactly* on the border is close to zero.

Or, to put it another way: the margin of error introduced by simplified boundaries is orders of magnitude higher

Reply:
Yes fully agree. I think I just left this comment there because in the corresponding spatial rule test we had a check like this. Will remove the comment.